packages/askar/src/storage/AskarStorageService.ts

```diff
@@ -0,0 +1,291 @@
+      await session.remove({ category: recordClass.type, name: id })
+    } catch (error) {
+      if (isAskarError(error) && error.code === askarErrors.NotFound) {
+        throw new RecordNotFoundError(`record with id ${id} not found.`, {
+          recordType: recordClass.type,
+          cause: error,
+        })
+      }
+      throw new WalletError('Error deleting record', { cause: error })
+    }
+  }
+
+  /** @inheritDoc */
+  public async getById(agentContext: AgentContext, recordClass: BaseRecordConstructor<T>, id: string): Promise<T> {
+    assertAskarWallet(agentContext.wallet)
+    const session = (agentContext.wallet as AskarWallet).session
+
+    try {
+      const record = await session.fetch({ category: recordClass.type, name: id })
+      if (!record) {
+        throw new RecordNotFoundError(`record with id ${id} not found.`, {
+          recordType: recordClass.type,
+        })
+      }
+      return this.recordToInstance(record, recordClass)
+    } catch (error) {
+      if (
+        isAskarError(error) &&
+        (error.code === askarErrors.NotFound ||
+          // FIXME: this is current output from askar wrapper but does not describe specifically a not found scenario
+          error.message === 'Received null pointer. The native library could not find the value.')
+      ) {
+        throw new RecordNotFoundError(`record with id ${id} not found.`, {
+          recordType: recordClass.type,
+          cause: error,
+        })
+      }
+      throw new WalletError(`Error getting record`, { cause: error })
+    }
+  }
+
+  /** @inheritDoc */
+  public async getAll(agentContext: AgentContext, recordClass: BaseRecordConstructor<T>): Promise<T[]> {
+    assertAskarWallet(agentContext.wallet)
+    const session = (agentContext.wallet as AskarWallet).session
+
+    const records = await session.fetchAll({ category: recordClass.type })
+
+    const instances = []
+    for (const record of records) {
+      instances.push(this.recordToInstance(record, recordClass))
+    }
+    return instances
+  }
+
+  /** @inheritDoc */
+  public async findByQuery(
+    agentContext: AgentContext,
+    recordClass: BaseRecordConstructor<T>,
+    query: Query<T>
+  ): Promise<T[]> {
+    assertAskarWallet(agentContext.wallet)
+    const store = agentContext.wallet.store
+
+    const askarQuery = this.askarQueryFromSearchQuery(query)
+
+    const scan = new Scan({
+      category: recordClass.type,
+      store,
+      tagFilter: askarQuery,
+    })
+
+    const instances = []
+    try {
+      const records = await scan.fetchAll()
+      for (const record of records) {
+        instances.push(this.recordToInstance(record, recordClass))
+      }
+      return instances
+    } catch (error) {
+      if (
+        isAskarError(error) && // FIXME: this is current output from askar wrapper but does not describe specifically a 0 length scenario
+        error.message === 'Received null pointer. The native library could not find the value.'
```

Comment:
Isn't this something we can fix in the wrapper? So the wrapper should jus allow a null pointer as return value? Maybe @blu3beri knows

Reply:
Yeah if that is the intended behaviour we can fix that in the wrapper.

Reply:
I've added https://github.com/hyperledger/aries-askar/issues/77 to follow this issue. I'm not absolutely sure that is a specific problem for JS, as a preliminar quick-debugging I've done showed me that the wrapper is getting an Error Code: 0 in those cases so it tries to create an EntryList from a null pointer (and that's why it shows that custom error).